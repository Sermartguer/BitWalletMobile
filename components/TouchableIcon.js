import React from 'react'
import {TouchableOpacity} from 'react-native'
import { MaterialCommunityIcons } from '@expo/vector-icons'

const TouchableIcon = ({ margin, containerStyle, onPress, size, name, color, ...otherProps }) => (
    <TouchableOpacity style={[{ margin }, containerStyle]} onPress={onPress}>
        <MaterialCommunityIcons size={size} name={name} color={color} />
    </TouchableOpacity>
)

TouchableIcon.defaultProps = {
    containerStyle: {},
    margin: 0,
    size: 22,
    name: '',
    color: '#111'
}

export default TouchableIcon