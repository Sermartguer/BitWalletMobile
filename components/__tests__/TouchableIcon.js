import 'react-native'
import React from 'react'
import TouchableIcon from '../TouchableIcon'
import renderer from 'react-test-renderer'

test('should render correctly', () => {
    const tree = renderer.create(<TouchableIcon />).toJSON()
    expect(tree).toMatchSnapshot()
})
