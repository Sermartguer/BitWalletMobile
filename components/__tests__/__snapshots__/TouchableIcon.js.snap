// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "margin": 0,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  />
</View>
`;
